csharp/src/Ice/TraceUtil.cs

```diff
@@ -2,315 +2,190 @@
+            TraceResponse(
+                "sending response",
+                communicator,
+                header,
+                frame.ReplyStatus,
+                frame.Encoding);
+
+        internal static void TraceFrame(Communicator communicator, Span<byte> header, IncomingRequestFrame frame) =>
+            TraceRequest(
+                "received request",
+                communicator,
+                header,
+                frame.Identity,
+                frame.Facet,
+                frame.Operation,
+                frame.IsIdempotent,
+                frame.Context,
+                frame.Encoding);
+
+        internal static void TraceFrame(Communicator communicator, Span<byte> header, IncomingResponseFrame frame) =>
+            TraceResponse(
+                "received response",
+                communicator,
+                header,
+                frame.ReplyStatus,
+                frame.Encoding);
+
+        private static void TraceRequest(
+            string heading,
+            Communicator communicator,
+            Span<byte> header,
+            Identity identity,
+            string facet,
+            string operation,
+            bool isIdempotent,
+            IReadOnlyDictionary<string, string> context,
+            Encoding encoding)
         {
             if (communicator.TraceLevels.Protocol >= 1)
             {
-                var iss = new InputStream(communicator, Ice1Definitions.Encoding,
-                    buffer.GetSegment(0, buffer.GetByteCount()).Array!);
-
-                using var s = new System.IO.StringWriter(CultureInfo.CurrentCulture);
-                Ice1Definitions.FrameType type = PrintMessage(s, iss);
-
-                communicator.Logger.Trace(communicator.TraceLevels.ProtocolCat,
-                    "sending " + GetFrameTypeAsString(type) + " " + s.ToString());
-            }
-        }
-
-        internal static void TraceRecv(Communicator communicator, System.ArraySegment<byte> buffer)
-        {
-            if (communicator.TraceLevels.Protocol >= 1)
-            {
-                var iss = new InputStream(communicator, Ice1Definitions.Encoding, buffer);
-
-                using var s = new System.IO.StringWriter(CultureInfo.CurrentCulture);
-                Ice1Definitions.FrameType type = PrintMessage(s, iss);
-
-                communicator.Logger.Trace(communicator.TraceLevels.ProtocolCat,
-                    "received " + GetFrameTypeAsString(type) + " " + s.ToString());
-            }
-        }
-
-        internal static void Trace(string heading, Communicator communicator, System.ArraySegment<byte> buffer)
-        {
-            if (communicator.TraceLevels.Protocol >= 1)
-            {
-                var iss = new InputStream(communicator, Ice1Definitions.Encoding, buffer);
-
-                using var s = new System.IO.StringWriter(CultureInfo.CurrentCulture);
+                using var s = new StringWriter(CultureInfo.CurrentCulture);
```

Comment:
Why do we use a StringWriter instead of a plain StringBuilder?
According to the doc,  `using` with StringWriter is not needed since it's Dispose does nothing.

Reply:
Fixed